app/models/concerns/has_wise_recipient.rb

```diff
@@ -0,0 +1,161 @@
+        fields << ACCOUNT_NUMBER_FIELD
+      elsif currency == "LKR"
+        fields << { type: :text_field, key: "branch_name", label: "Branch name" }
+        fields << ACCOUNT_NUMBER_FIELD
+      elsif currency == "MAD"
+        fields << { type: :text_field, key: "account_number", label: "Account number", description: "24-digit RIB number" }
+      elsif currency == "MXN"
+        fields << { type: :text_field, key: "account_number", label: "Account number", description: "18-digit CLABE number" }
+        fields << { type: :text_field, key: "tax_id", label: "CURP number", description: "18-character CURP" }
+      elsif currency == "MYR"
+        fields << { type: :select, key: "account_type", label: "Account type", options: { "Mobile Number (Duitnow)": "mobile_number_duitnow", "NRIC (Duitnow)": "nirc_duitnow", "BNR (Duitnow)": "bnr_duitnow", "Bank Account": "bank_account" } }
+        fields << { type: :text_field, key: "duitnow_id", label: "Duitnow ID", conditional: "account_type != 'bank_account'" }
+        fields << { type: :text_field, key: "account_number", label: "Account number", conditional: "account_type == 'bank_account'" }
+      elsif currency == "UYU"
+        fields << { type: :select, key: "account_type", label: "Account type", options: { "Checking": "checking", "Savings": "savings" } }
+        fields << ACCOUNT_NUMBER_FIELD
+        fields << { type: :text_field, key: "tax_id", label: "National ID" }
+        fields << { type: :text_field, key: "branch_name", label: "Branch name (if applicable)" }
+      elsif currency == "VND"
+        fields << { type: :text_field, key: "institution_number", label: "Bank code (BIC/SWIFT)", placeholder: "AAAA-BB-CC-123" }
+        fields << ACCOUNT_NUMBER_FIELD
+      elsif currency == "ZAR"
+        fields << { type: :date_field, key: "recipient_birthday", label: "Recipient's date of birth" }
+        fields << ACCOUNT_NUMBER_FIELD
+      end
+      return fields
+    end
+
+    def self.recipient_information_keys
+      fields = []
+      AVAILABLE_CURRENCIES.each do |currency|
+        fields += self.information_required_for(currency)
+      end
+      fields.collect{ |field| field[:key] }.uniq
+    end
+
+    def self.recipient_information_accessors
+      recipient_information_keys - %w[account_number institution_number branch_number tax_id recipient_birthday]
+    end
+
+    store_accessor :recipient_information, *self.recipient_information_accessors
+  end
+
+  # IBAN & postal code formats sourced from https://column.com/docs/international-wires/country-specific-details
+
+  POSTAL_CODE_FORMATS = {
+    "US": /\A\d{5}(?:-\d{4})?\z/,
+    "CN": /\A\d{6}\z/,
+    "JP": /\A\d{3}-\d{4}\z/,
+    "FR": /\A\d{5}\z/,
+    "DE": /\A\d{5}\z/
+  }.freeze
+
+  ACCOUNT_NUMBER_FIELD = { type: :text_field, key: "account_number", placeholder: "123456789", label: "Account number" }.freeze
+
+  AVAILABLE_CURRENCIES = (::EuCentralBank::CURRENCIES + ["EUR"] - ["USD"]).freeze
```

Comment:
Where did we get this list from? This was what we used for wires only because this is what we could convert